src/app/modules/main/module.nim

```diff
@@ -662,3 +663,21 @@ method mnemonicBackedUp*[T](self: Module[T]) =
     conf.SETTINGS_SECTION_ID,
     self.calculateProfileSectionHasNotification(),
     notificationsCount = 0)
+
+method osNotificationClicked*[T](self: Module[T], details: NotificationDetails) =
+  if(details.notificationType == NotificationType.NewContactRequest):
+    self.controller.switchTo(details.sectionId, "", "")
+    self.view.emitOpenContactRequestsPopupSignal()
+  elif(details.notificationType == NotificationType.JoinCommunityRequest):
+    self.controller.switchTo(details.sectionId, "", "")
+    self.view.emitOpenCommunityMembershipRequestsPopupSignal(details.sectionId)
+  elif(details.notificationType == NotificationType.MyRequestToJoinCommunityAccepted):
+    self.controller.switchTo(details.sectionId, "", "")
+  elif(details.notificationType == NotificationType.MyRequestToJoinCommunityRejected):
+    info "There is no particular action clicking on a notification informing you about rejection to join community"
+
+method newCommunityMembershipRequestReceived*[T](self: Module[T], membershipRequest: CommunityMembershipRequestDto) =
+  let (contactName, _, _) = self.controller.getContactNameAndImage(membershipRequest.publicKey)
+  let community =  self.controller.getCommunityById(membershipRequest.communityId)
+  singletonInstance.globalEvents.newCommunityMembershipRequestNotification("New membership request",
+  fmt "{contactName} asks to join {community.name}", community.id)
```

Comment:
Would there be a way for QML to handle the text, since we can't translate those here?

Reply:
@jrainville yes, that's something I mentioned in the first comment here. We should find a way how to handle that, but this is not the only place where we have some "text" on the nim side. So would be good to have a ticket for handling that, and sort that out everywhere anyhow (actually finding a solution) and once we completely move to c++ there will be lot easier.